proxy-robots-module/src/main/java/orwell/proxy/udp/UdpBeaconFinder.java

```diff
@@ -0,0 +1,161 @@
+package orwell.proxy.udp;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.*;
+import java.util.Enumeration;
+
+/**
+ * Created by Michaël Ludmann on 5/25/15.
+ */
+public class UdpBeaconFinder {
+    private final static Logger logback = LoggerFactory.getLogger(UdpBeaconFinder.class);
+    private final int broadcastPort;
+    private final int receiverBufferSize = 512;
+    private final DatagramSocket datagramSocket;
+    private final UdpBeaconDecoder udpBeaconDecoder;
+    private int maxAttemptsNumber;
+    private int attemptsPerformed = 0;
+    private String pushAddress;
+    private String subscribeAddress;
+
+    public UdpBeaconFinder(final DatagramSocket datagramSocket,
+                           final int broadcastPort,
+                           final UdpBeaconDecoder udpBeaconDecoder) {
+        this.datagramSocket = datagramSocket;
+        this.broadcastPort = broadcastPort;
+        this.udpBeaconDecoder = udpBeaconDecoder;
+        maxAttemptsNumber = 1;
+    }
+
+    /**
+     * Change number of attempts to find the UDP beacon server
+     * Default value is set to 1
+     * @param maxAttemptsNumber
+     */
+    public void setMaxAttemptsNumber(final int maxAttemptsNumber) {
+        if (0 > maxAttemptsNumber) {
+            logback.warn("Udp broadcasting attempts number cannot be less than 0");
+        } else {
+            this.maxAttemptsNumber = maxAttemptsNumber;
+        }
+    }
+
+    public int getNumberOfPerformedAttempts() {
+        return attemptsPerformed;
+    }
+
+    /**
+     * @return true if we did not get any meaningful response from server
+     * and have not reached max allowed attempts number
+     */
+    private boolean shouldTryToFindBeacon() {
+        if (null == udpBeaconDecoder) {
```

Comment:
You are right, I will add an assert != null in the constructor and remove the other checks.